sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/routing/RoutingMapProviderHelperTest.java

```diff
@@ -126,4 +133,56 @@ public String apply(PartitionKeyRange range) {
 
         assertThat("3,4").isEqualTo(ranges.stream().map(func).collect(Collectors.joining(",")));
     }
+
+    @Test(groups = {"unit"})
+    public void getOverlappingRangesWithList() {
+
+        Function<PartitionKeyRange, String> func = new Function<PartitionKeyRange, String>() {
+            @Override
+            public String apply(PartitionKeyRange range) {
+                return range.getId();
+            }
+        };
+
+        IRoutingMapProvider routingMapProviderMock = Mockito.mock(IRoutingMapProvider.class);
+        List<PartitionKeyRange> rangeList = Arrays.asList(new PartitionKeyRange("0", "", "000A"),
+                                                          new PartitionKeyRange("1", "000A", "000D"),
+                                                          new PartitionKeyRange("2", "000D", "0012"),
+                                                          new PartitionKeyRange("3", "0012", "0015"),
+                                                          new PartitionKeyRange("4", "0015", "0020"),
+                                                          new PartitionKeyRange("5", "0020", "0040"),
+                                                          new PartitionKeyRange("6", "0040", "FF"));
+        Mono<List<PartitionKeyRange>> listSingle = Mono.just(rangeList);
+
+        Map<Range, List<PartitionKeyRange>> resultMap = new HashMap<>();
+
+        resultMap.put(new Range<>("000D", "0012", true, false),
+                      Collections.singletonList(new PartitionKeyRange("2", "000D", "0012")));
+        resultMap.put(new Range<>("0012", "0015", true, false),
+                      Collections.singletonList(new PartitionKeyRange("3", "0012", "0015")));
+        resultMap.put(new Range<>("0015", "0020", true, false),
+                      Collections.singletonList(new PartitionKeyRange("4", "0015", "00120")));
+
+        Mockito.doAnswer(invocationOnMock -> {
+            Range range = invocationOnMock.getArgumentAt(1, Range.class);
+            return Mono.just(new Utils.ValueHolder<>(resultMap.get(range)));
+        }).when(routingMapProviderMock).tryGetOverlappingRangesAsync(Matchers.anyString(),
+                                                                     Matchers.any(),
+                                                                     Matchers.anyBoolean(),
+                                                                     Matchers.anyMap());
+
+        Mono<List<PartitionKeyRange>> overlappingRanges;
+        overlappingRanges = RoutingMapProviderHelper.getOverlappingRanges(routingMapProviderMock,
+                                                                          "/dbs/db1/colls/coll1",
```

Comment:
generally a collection self link is passed here not name link.
I know it is a test but to avoid future confusion I would update this to look like a self link.
not a big deal really.  :-)

Reply:
This expects a resourceId, changed the test parameter to a id to avoid confusion. Thanks